src/ng-pattern-restrict.js

```diff
@@ -238,10 +230,10 @@ angular.module('ngPatternRestrict', [])
 
           function detectSetCaretPositionMethods() {
             var input = iElement[0];
-            if (notThrows(function () { input.setSelectionRange(0, 0); })) {
+            if (typeof(input.setSelectionRange) === 'function') {
               setCaretPosition = setCaretPositionWithSetSelectionRange;
-            } else if (notThrows(function () { return input.createTextRange(); })) {
```

Comment:
Hi! Firstly, thanks a lot for the changes -- I really appreciate you helping out.

As for the notThrows() check, this was in order to check that createTextRange was available, since the browser would report that it is present and that it is a function but will actually throw an exception when invoked.

I'll try now to identify which browser was this so that you can check if this solution works in that browser as well.